src/Rules.UnitTests/EmbeddedResourceTests.cs

```diff
@@ -58,6 +61,44 @@ public void CheckEmbeddedDescriptionFiles_ByLanguage(string languageKey, int atL
             resourceNames.Should().HaveCountGreaterThan(atLeastXRules);
         }
 
+        [TestMethod]
+        public void CheckEmbeddedDescriptionFiles_AreParseableAsXml()
+        {
+            // Performance: this test is loading and parsing nearly 2000 files,
+            // but is still only takes a few hundred milliseconds.
+            var asm = typeof(RuleMetadataProvider).Assembly;
+            var resourceNames = asm.GetManifestResourceNames()
+                .Where(x => x.EndsWith(".desc"));
+
+            Console.WriteLine("Checking embedded files. Count = " + resourceNames.Count());
+            var failures = resourceNames.Where(x => !ReadResourceAsXml(asm, x))
+                .ToArray();
+
+            failures.Should().HaveCount(0);
+        }
+
+        private static bool ReadResourceAsXml(Assembly asm, string fullResourceName)
+        {
+            try
+            {
+                var readerSettings = new XmlReaderSettings
+                {
+                    ConformanceLevel = ConformanceLevel.Fragment
+                };
+                using var stream = new StreamReader(asm.GetManifestResourceStream(fullResourceName));
+                using var reader = XmlReader.Create(stream, readerSettings);
```

Comment:
The `XmlReader` is very fast, so parsing 2000 files in the test only takes a few hundred milliseconds.